pkg/cloud/services/compute/instance.go

```diff
@@ -481,6 +484,16 @@ func createPort(is *Service, clusterName string, name string, net *infrav1.Netwo
 	return *newPort, nil
 }
 
+func deletePorts(s *Service, nets []servers.Network) error {
+
+	for _, n := range nets {
+		if err := ports.Delete(s.networkClient, n.Port).ExtractErr(); err != nil {
```

Comment:
just to confirm, this delete action already handle the NOT EXIST error or not ?

Reply:
This function is executed just after checking port existence or creating  as of now. but OK, I will add checking existence.